identity/src/main/java/com/android/identity/credential/PendingAuthenticationKey.kt

```diff
@@ -0,0 +1,227 @@
+ * the issuing server for outstanding certification requests.
+ *
+ *
+ * Once certification is complete, use [.certify]
+ * to upgrade to a [AuthenticationKey].
+ */
+class PendingAuthenticationKey {
+    /**
+     * The secure area for the pending authentication key.
+     *
+     * This can be used together with the alias returned in [alias].
+     */
+    lateinit var secureArea: SecureArea
+        private set
+
+    /**
+     * The alias for the pending authentication key.
+     *
+     * This can be used together with the [SecureArea] returned by [secureArea].
+     */
+    lateinit var alias: String
+        private set
+
+    /**
+     * The domain of the pending authentication key.
+     *
+     * This returns the domain set when the pending authentication key was created.
+     */
+    lateinit var domain: String
+        private set
+
+    /**
+     * Application specific data.
+     *
+     * Use this object to store additional data an application may want to associate
+     * with the pending authentication key. Setters and associated getters are
+     * enumerated in the [ApplicationData] interface.
+     */
+    internal lateinit var privateApplicationData: SimpleApplicationData
+    val applicationData: ApplicationData
+        get() = privateApplicationData
+
+    /**
+     * Gets the authentication key counter.
+     *
+     *
+     * This is the value of the Credential's Authentication Key Counter
+     * at the time this pending authentication key was created.
+     *
+     * @return the authentication key counter.
+     */
+    var authenticationKeyCounter: Long = 0
+        private set
+
+    val attestation: List<X509Certificate>
+        /**
+         * The X.509 certificate chain for the authentication key pending certification.
+         *
+         * The application should send this key to the issuer which should create issuer-provided
+         * data (e.g. an MSO if using ISO/IEC 18013-5:2021) using the key as the `DeviceKey`.
+         *
+         * @return An X.509 certificate chain for the pending authentication key.
+         */
+        get() {
+            return secureArea.getKeyInfo(alias).attestation
+        }
+
+    /**
+     * The auth key that will be replaced by this key once it's been certified.
+     *
+     * @return An [AuthenticationKey] or `null` if no key was designated
+     * when this pending key was created.
+     */
+    val replacementFor: AuthenticationKey?
+        get() {
+            if (replacementForAlias == null) {
+                return null
+            }
+            for (authKey in credential.authenticationKeys) {
+                if (authKey.alias == replacementForAlias) {
+                    return authKey
+                }
+            }
+            Logger.w(TAG, "Key with alias $replacementForAlias which " +
+                    "is intended to be replaced does not exist")
+            return null
+        }
```

Comment:
we can use `firstOrNull { }` to return `authKey` if there's a matching `authKey.alias`. 

To add logging we can use the `also` scope function

```
get() = credential.authenticationKeys.firstOrNull { it.alias == replacementForAlias }
            .also {
                if (it == null && replacementForAlias != null) {
                    Logger.w(
                        TAG, "Key with alias $replacementForAlias which " +
                                "is intended to be replaced does not exist"
                    )
                }
            }
```